container/lxd/network.go

```diff
@@ -0,0 +1,181 @@
+package lxd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/lxc/lxd/shared/api"
+
+	"github.com/juju/juju/network"
+)
+
+// LocalBridgeName returns the name of the local LXD network bridge.
+func (s *Client) LocalBridgeName() string {
+	return s.localBridgeName
+}
+
+func (s *Client) VerifyDefaultBridge(profile *api.Profile) error {
+	eth0, ok := profile.Devices["eth0"]
+	if !ok {
+		// On LXD >= 2.3 there is no bridge config by default.
+		// Ensure that the network bridge and eth0 device both exist.
+		if s.networkAPISupport {
+			return errors.Annotate(s.ensureDefaultBridge(profile), "ensuring default bridge config")
+		}
+		return errors.Errorf("unexpected LXD %q profile without eth0 device: %+v", profile.Name, profile)
+	}
+
+	netName := eth0["parent"]
+
+	if s.networkAPISupport {
+		net, _, err := s.GetNetwork(netName)
+		if err != nil {
+			return errors.Annotatef(err, "retrieving network %q", netName)
+		}
+		if err := ensureNoIPv6(net); err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	if eth0["type"] != "nic" || eth0["nictype"] != "bridged" {
+		return errors.Errorf("eth0 is not configured as a bridge in %q profile: %v", profile.Name, eth0)
+	}
+
+	s.localBridgeName = netName
+	logger.Infof("LXD %q profile uses network bridge %q", profile.Name, netName)
+
+	// If this LXD version supports the network API, we are done.
+	// Otherwise we need to check the legacy lxd-bridge config file.
+	if s.networkAPISupport {
+		return nil
+	}
+	return errors.Trace(checkBridgeConfigFile(ioutil.ReadFile))
+}
+
+// ensureDefaultBridge ensures that the default LXD bridge exists,
+// that it is not configured to use IPv6, and that the eth0 device exists in
+// the input profile.
+// An error is returned if the bridge exists with IPv6 configuration.
+// If the bridge does not exist, it is created.
+func (s *Client) ensureDefaultBridge(profile *api.Profile) error {
+	net, eTag, err := s.GetNetwork(network.DefaultLXDBridge)
+	if err != nil {
+		if !isLXDNotFound(err) {
+			return errors.Trace(err)
+		}
+		req := api.NetworksPost{
+			Name: network.DefaultLXDBridge,
+			NetworkPut: api.NetworkPut{Config: map[string]string{
+				"ipv6.address": "none",
+				"ipv6.nat":     "false",
+			}},
+		}
+		err := s.CreateNetwork(req)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		net, eTag, err = s.GetNetwork(network.DefaultLXDBridge)
```

Comment:
I'd also ultimately really like it if we didn't depend on 'lxdbr0' to exist, but had the concept of spaces for LXD and one of those could be the 'lxdbr0' bridge.
But that doesn't have to happen in this patch.